Java/syntax_test_java.java

```diff
@@ -312,3 +371,502 @@ void test2() {
 //            ^ support.class
     }
 }
+
+public enum FooEnum {
+//     ^^^^ storage.type.java
+  FOO;
+//^^^ constant.other.enum
+}
+// <- meta.class.java meta.class.body.java punctuation.section.class.end.java
```

Comment:
When possible in more recent syntax changes, we are trying not to "stack" scopes. So instead of having `meta.class.java meta.class.body.java` we have `public enum FooEnum ` as `meta.class.java` and `{…}` as `meta.class.body.java`. Does that make sense? It can take some extra context commands, but in many languages it can help prevent really deep scope stacks.